lang_js/parsing/ast_js.ml

```diff
@@ -213,7 +213,7 @@ and expr =
   (* ident is None when assigned in module.exports  *)
   | Class of class_definition * a_ident option
 
-  | ObjAccess of expr * tok * property_name
+  | ObjAccess of expr * tok * (* is optional `?.` *) bool * property_name
```

Comment:
maybe I would have made a variant here.

BTW here it's totally fine to have a dot_operator, because in those language-specific ASTs it's good to be
as close as possible to the original code. If we do some unsugaring, we usually prefer to do it
when we translate to the generic AST.